gcc/config/loongarch/constraints.md

```diff
@@ -113,6 +113,13 @@
 (define_register_constraint "j" "SIBCALL_REGS"
   "@internal")
 
+(define_memory_constraint "k"
+  "A memory operand whose address is formed by a base register and offset.
+   And offset is register or 12 bit immediate."
+  (and (match_code "mem")
+       (not (match_test "loongarch_14bit_shifted_offset_address_p (XEXP (op, 0), mode)"))
+       (not (match_test "loongarch_12bit_offset_address_p (XEXP (op, 0), mode)"))))
```

Comment:
Judging from the code below this does seem like intended for `ldx/stx`, and not `alsl`, and I don't get why there's need to restrict the range to 14bit and 12bit.